sogs/model/message.py

```diff
@@ -0,0 +1,94 @@
+from .. import db
+from ..db import insert_and_get_pk, query
+
+from .user import User
+
+
+class Message:
+    """Class representing a DM between users
+
+    Properties:
+        sender: sender user of the message
+        recip: recipant user of the message
+        data: opaque message data
+        signature: signature of data
+    """
+
+    def __init__(self, row=None, *, sender=None, recip=None, data=None, signature=None):
+        """
+        Constructs a Message from a pre-retrieved row *or* sender recipiant data and signature.
+        """
+        if row is None:
+            if None in (sender, recip, data, signature):
+                raise ValueError("Message() error: no row or data provided")
+            if not all(isinstance(arg, User) for arg in (sender, recip)):
+                raise ValueError("Message() error: sender or recipient was not a User model")
+
+            with db.transaction():
+                row = insert_and_get_pk(
+                    """
+            INSERT INTO inbox (sender, recipient, data, signature)
+            VALUES (:sender, ....)
+            """,
+                    "id",
+                    sender=sender.id,
+                    recipiant=recip.id,
+                    data=data,
+                    signture=signature,
+                ).first()
+        # sanity check
+        assert row is not None
+        self._row = row
+
+    @staticmethod
+    def to(user, since=None):
+        """ get all message for a user, returns a generator """
+        if since:
+            rows = query(
+                "SELECT * FROM inbox WHERE recipient = :recip AND id > :since_id",
+                recip=user.id,
+                since_id=since,
+            )
+        else:
+            rows = query("SELECT * FROM inbox WHERE recipient = :recip", recip=user.id)
+        for row in rows:
+            yield Message(row=row)
+
+    @property
+    def info(self):
+        return {
+            "id": self.id,
+            "expires_at": self.expires_at,
+            "data": self.data,
+            "signature": self.signature,
+            "sender": self.sender.session_id,
+            "recipient": self.recipient.session_id,
```

Comment:
Is there any reason to include this field in the response to the message retriever?